File: ZoneSentry/ClientApp/src/api/models/RealtyServiceOrder.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { ApplicationUser } from './ApplicationUser';
import type { Realty } from './Realty';
import type { RealtyService } from './RealtyService';
import type { RealtyServicePayment } from './RealtyServicePayment';

export type RealtyServiceOrder = {
    id?: number;
    cost?: number;
    date?: string;
    realty?: Realty;
    realtyService?: RealtyService;
    payments?: Array<RealtyServicePayment> | null;
    companyOrdered?: boolean;
    orderer?: ApplicationUser;
};

File: ZoneSentry/Models/RealtyServiceOrder.cs
namespace ZoneSentry.Models;

public class RealtyServiceOrder
{
    public int Id { get; set; }
    
    public int Cost { get; set; }
    public DateTime Date { get; set; }

    public Realty Realty { get; set; }
    public RealtyService RealtyService { get; set; }
    
    public List<RealtyServicePayment> Payments { get; set; }
    
    public bool CompanyOrdered { get; set; }
    public ApplicationUser? Orderer { get; set; }
}

public class RealtyServiceOrderDTO
{
    public int Id { get; set; }
    
    public int Cost { get; set; }
    public DateTime Date { get; set; }
    public bool Active { get; set; }

    public RealtyUserView Realty { get; set; }
    public RealtyServiceDTO RealtyService { get; set; }
    
    public bool CompanyOrdered { get; set; }
    public ApplicationUserDTO? Orderer { get; set; }
}